src/EventStore.Client/SubscriptionPosition.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+
+#nullable enable
+namespace EventStore.Client {
+	/// <summary>
+	/// A structure representing the logical position of a subscription to all. />
+	/// </summary>
+	public readonly struct SubscriptionPosition : IEquatable<SubscriptionPosition>, IComparable<SubscriptionPosition> {
+		/// <summary>
+		/// Represents a <see cref="SubscriptionPosition"/> when no events have been seen (i.e., the beginning).
+		/// </summary>
+		public static readonly SubscriptionPosition Start = new(null);
+
+		/// <summary>
+		/// Represents a <see cref="SubscriptionPosition"/> to receive events written after the subscription is confirmed.
+		/// </summary>
+		public static readonly SubscriptionPosition Live = new(Position.End);
+
+		/// <summary>
+		/// Returns a <see cref="SubscriptionPosition"/> for the given <see cref="Position"/>.
+		/// </summary>
+		/// <param name="position">The <see cref="Position"/>.</param>
+		/// <returns></returns>
+		/// <exception cref="ArgumentException"></exception>
+		public static SubscriptionPosition After(Position position) => position == Position.End
+			? throw new ArgumentException("Use 'SubscriptionPosition.Live.'", nameof(position))
```

Comment:
**QUESTION:** Is this validation necessary? Maybe a warning would be enough?

Reply:
I think it is. Storing Live/End in your database doesn't make a lot of sense...

Reply:
That's fair 👍